File: ch36p1066peopleGroups/ch36p1066peopleGroups/PeopleDocument.m

#import "PeopleDocument.h"

@implementation PeopleDocument

-(id)initWithFileURL:(NSURL *)url {
    self = [super initWithFileURL:url];
    if (self) {
        self->_people = [NSMutableArray array];
    }
    return self;
}

- (BOOL)loadFromContents:(id)contents ofType:(NSString *)typeName error:(NSError **)outError {
    NSLog(@"loading %@", typeName);
    NSArray* arr = [NSKeyedUnarchiver unarchiveObjectWithData:contents];
    self.people = [NSMutableArray arrayWithArray:arr];
    return YES;
}

- (id)contentsForType:(NSString *)typeName error:(NSError **)outError {
    NSLog(@"archiving %@", typeName);
    NSData* data = [NSKeyedArchiver archivedDataWithRootObject:self.people];
    return data;
}

@end

File: iOS8bookExamples/bk2ch23p809peopleGroups/ch36p1066peopleGroups/PeopleDocument.swift


import UIKit

class PeopleDocument: UIDocument {
    
    var people = [Person]()
    
    override func loadFromContents(contents: AnyObject,
        ofType typeName: String,
        error outError: NSErrorPointer) -> Bool {
            println("loading \(typeName)")
            if let contents = contents as? NSData {
                let arr = NSKeyedUnarchiver.unarchiveObjectWithData(contents) as! [Person]
                self.people = arr
                return true
            }
            return false
    }
    
    override func contentsForType(typeName: String,
        error outError: NSErrorPointer) -> AnyObject? {
            println("archiving \(typeName)")
            let data = NSKeyedArchiver.archivedDataWithRootObject(self.people)
            return data
    }
    
}
